resharper/resharper-unity/src/Rider/Notifications/GeneratedFileNotification.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using JetBrains.Annotations;
+using JetBrains.Core;
+using JetBrains.Lifetimes;
+using JetBrains.Platform.Unity.EditorPluginModel;
+using JetBrains.ProjectModel;
+using JetBrains.ProjectModel.Tasks;
+using JetBrains.ReSharper.Features.Inspections.Bookmarks.NumberedBookmarks;
+using JetBrains.ReSharper.Features.XamlRendererHost.Preview;
+using JetBrains.ReSharper.Host.Features.Notifications;
+using JetBrains.ReSharper.Host.Features.ProjectModel;
+using JetBrains.ReSharper.Host.Features.TextControls;
+using JetBrains.ReSharper.Plugins.Unity.AsmDefNew.Psi.Caches;
+using JetBrains.Util;
+using JetBrains.Util.Extension;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Rider.Notifications
+{
+    [SolutionComponent]
+    public class GeneratedFileNotification
+    {
+        private readonly NotificationPanelHost myNotificationPanelHost;
+
+        public GeneratedFileNotification(Lifetime lifetime, UnityHost unityHost, ConnectionTracker connectionTracker, UnityEditorProtocol editorProtocol, ISolution solution,
+            AsmDefNameCache asmDefNameCache, [CanBeNull] TextControlHost textControlHost = null, [CanBeNull] SolutionLifecycleHost solutionLifecycleHost = null,
+            [CanBeNull] NotificationPanelHost notificationPanelHost = null)
+        {
+            if (solutionLifecycleHost == null)
+                return;
+            
+            myNotificationPanelHost = notificationPanelHost;
+                
+            solutionLifecycleHost.FullStartupFinished.Advise(lifetime, _ =>
```

Comment:
These subscriptions will happen for non-Unity projects. Add a check for Unity solution somewhere?